visualization/dashboard/src/MLIDashboard/Controls/CohortEditor/CohortEditor.tsx

```diff
@@ -271,34 +298,50 @@ export class CohortEditor extends React.PureComponent<ICohortEditorProps, ICohor
         )
     }
 
-    private readonly setNumericValue = (delta: number, column: IJointMeta, stringVal: string): string | void => {
+    private readonly setNumericValue = (delta: number, column: IJointMeta, index=0, stringVal:string): string | void => {
         const openedFilter = this.state.openedFilter;
         if (delta === 0) {
-            const number = +stringVal;
-            if ((!Number.isInteger(number) && column.featureRange.rangeType === RangeTypes.integer)
-                || number > column.featureRange.max || number < column.featureRange.min) {
+            const numberVal = +stringVal;
+            if ((!Number.isInteger(numberVal) && column.featureRange.rangeType === RangeTypes.integer)
+                || numberVal > column.featureRange.max || numberVal < column.featureRange.min) {
                 return this.state.openedFilter.arg.toString();
             }
-
-            this.setState(
-                {
-                    openedFilter: {
-                        arg: number,
-                        method: openedFilter.method,
-                        column: openedFilter.column
+            if(openedFilter.method === FilterMethods.inTheRangeOf){
```

Comment:
Is the check needed? It seems you've handled this by setting the default index to 0?